library/default-synths-extra.scd

```diff
@@ -523,3 +523,157 @@ SynthDef(\superreese,  {|out, pan, freq, sustain, accelerate, detune=0, voice=0|
 	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env));
 }).add;
 );
+
+(
+// DrumSynths SC Example - SOS Drums by Renick Bell, renick@gmail.com
+// recipes from Gordon Reid in his Sound on Sound articles
+// SOSkick — http:www.soundonsound.com/sos/jan02/articles/synthsecrets0102.asp
+// increase modFreq and modIndex for interesting electronic percussion
+SynthDef(\soskick, {
+  arg out, pan, freq = 50, modFreq = 5, modIndex = 5, sustain = 1, beaterNoiseLevel = 0.025;
+
+  var pitchContour, drumOsc, drumLpf, drumEnv, env;
+  var beaterSource, beaterHpf, beaterLpf, lpfCutoffContour, beaterEnv;
+  var kickMix, snd;
+
+  pitchContour = Line.kr(freq * 2, freq, 0.02);
+  drumOsc = PMOsc.ar(pitchContour, modFreq, modIndex / 1.3, mul: 1, add: 0);
+  drumLpf = LPF.ar(in: drumOsc, freq: 1000, mul: 1, add: 0);
+  env = EnvGen.kr(Env.perc(0.005, sustain), 1.0, doneAction: Done.freeSelf);
+  drumEnv = drumLpf * env;
+
+  beaterSource = WhiteNoise.ar(beaterNoiseLevel);
+  beaterHpf = HPF.ar(in: beaterSource, freq: 500, mul: 1, add: 0);
+
+  lpfCutoffContour = Line.kr(6000, 500, 0.03);
+
+  beaterLpf = LPF.ar(in: beaterHpf, freq: lpfCutoffContour, mul: 1, add: 0);
+  beaterEnv = beaterLpf * EnvGen.kr(Env.perc, 1.0);
+
+  kickMix = Mix.new([drumEnv, beaterEnv]) * 2;
+
+  OffsetOut.ar(out, DirtPan.ar(kickMix, ~dirt.numChannels, pan, env));
+}, metadata: (
+  credit: "Renick Bell",
+  category: \drums,
+  tags: [\pitched, \bass, \sos]
+)).add;
+);
+
+(
+// Retrieved from http://sccode.org/1-5aD
+// DrumSynths SC Example - SOS Drums by Renick Bell, renick@gmail.com
+// Recipes from Gordon Reid in his Sound on Sound articles
+// SOShats — http:www.soundonsound.com/sos/Jun02/articles/synthsecrets0602.asp
+// Modified by Bruno Ruviaro and Josh Mitchell 8/19
+SynthDef(\soshats, {
+  arg
+  // Standard Values
+  out = 0, pan = 0, freq = 6000, amp = 0.8, curve = -4, rootIndex = 238.5, rq = 1, sustain = 0.1,
```

Comment:
For this and the other synths, I believe Tidal+SuperDirt will set certain parameter defaults before the SynthDef is even called.  For example, if in Tidal you use `d1 $ s "soshats"` then freq will be automatically set to 261.6 (C5), amp to 0.4, sustain to 1 (depending on the pattern and cps), and pan to 0.5.  So this synth is probably going to sound weird unless you do `d1 $ s "soshats/10" # n "fs9"` or something, which is maybe not the intent.

Reply:
Yeah, you're totally right. TidalCycles sets these parameters by default, but all the synths actually sound pretty nice with default TC's parameters too.

The intent to leave freq and amp was to make these parameters available from TC controls and to be them as predictable as possible. I don't really think that SuperDirt's synths must sound incredible with no parameters provided to them from the end-user, it's kinda tricky though.

Maybe you have some suggestions on how to resolve this issue?

Reply:
(I'm going to use `soshats` as the main example since it's the most extreme case, but these apply to the other drums too)

- amp

It's only 6 dB so I don't know how much it matters. It could either be left as is, or you could just multiply by the factor of two at the end of the SynthDef
- freq

One method would be to multiply any incoming freq by 20 (or 16 if you like powers of two) so that the default `n` results in something close to 6000 Hz. This has the advantage that the default sounds close to as intended but the user can still mess around with `n` to get different pitches.

If there's really no discernible pitch though, another option would be to rename the parameter entirely so that `n` doesn't affect it, hardcode in the 6000 Hz, and possibly allow rescaling using `pitch1` (already defined in Tidal but rarely used) or a similar parameter.   For example, use `6000 * pitch1` anywhere `freq` is currently used.

- sustain

Sustain is a bit tricky since SuperDirt goes to a lot of effort to automatically provide a sustain to "fit" the duration into the "slot" coming from Tidal, which is great for something like a piano, but not really how drums usually work.  Do you really want the hat to decay twice as fast if someone does `s "soshats*8"` instead of `s "soshats*4"`?  All that said, I don't know how noticeable it really is in the end - you're already using fixed attack times which will help things sound consistent.  It might be worth experimenting a little bit with really fast/slow sequences to see if you might want to wrap the release time in a `clip`, which would keep things from getting too weird at the extremes.

Reply:
@bgold-cosmos Thanks for your suggestions!

* amp
I've decided to drop out this parameter from synthdefs and completely rely on SuperDirt's env and gain controls.

* freq
I do think that frequency in drums makes a big difference and the main issue with multiplying its value by powers of two or any other number is kinda confusing when you're using `note "e3"` in TC, for example. The user's intention is to hear exactly `e3` note, but in the case of multiplication, he will hear a whole different pitch. 

* sustain
I think it's the same with `freq` param. Why should we stop the users from getting freaky? 

---

Actually, thanks a lot for reminding me about `pitch1` and also `voice` parameters, which we can use for synth's modifications. I will prepare all the synths in a while.

Reply:
re freq and sustain parameters: the difficulty is only to have good defaults, and good (that is interesting and somewhat transparent) deviations from defaults when something is specified. Sometimes you may want to interpret freq and sustain in a particular way from within a SynthDef.

Sometimes I write:

```supercollider
SynthDef(\x, { |freq = 440, speed = 1|
   var ratio = freq / 440;
   var trig = speed * ratio;
[…]
})
```